lib/msal-browser/src/interaction_client/NativeInteractionClient.ts

```diff
@@ -0,0 +1,314 @@
+ * Licensed under the MIT License.
+ */
+
+import { AuthenticationResult, Logger, ICrypto, PromptValue, AuthToken, Constants, AccountEntity, AuthorityType, ScopeSet, TimeUtils, AuthenticationScheme, UrlString, OIDC_DEFAULT_SCOPES, PopTokenGenerator, SignedHttpRequestParameters, IPerformanceClient } from "@azure/msal-common";
+import { BaseInteractionClient } from "./BaseInteractionClient";
+import { BrowserConfiguration } from "../config/Configuration";
+import { BrowserCacheManager } from "../cache/BrowserCacheManager";
+import { EventHandler } from "../event/EventHandler";
+import { PopupRequest } from "../request/PopupRequest";
+import { SilentRequest } from "../request/SilentRequest";
+import { SsoSilentRequest } from "../request/SsoSilentRequest";
+import { NativeMessageHandler } from "../broker/nativeBroker/NativeMessageHandler";
+import { NativeExtensionMethod, ApiId, TemporaryCacheKeys } from "../utils/BrowserConstants";
+import { NativeExtensionRequestBody, NativeTokenRequest } from "../broker/nativeBroker/NativeRequest";
+import { NativeResponse } from "../broker/nativeBroker/NativeResponse";
+import { NativeAuthError } from "../error/NativeAuthError";
+import { RedirectRequest } from "../request/RedirectRequest";
+import { NavigationOptions } from "../navigation/NavigationOptions";
+import { INavigationClient } from "../navigation/INavigationClient";
+import { BrowserAuthError } from "../error/BrowserAuthError";
+
+export class NativeInteractionClient extends BaseInteractionClient {
+    protected apiId: ApiId;
+    protected accountId: string;
+    protected nativeMessageHandler: NativeMessageHandler;
+
+    constructor(config: BrowserConfiguration, browserStorage: BrowserCacheManager, browserCrypto: ICrypto, logger: Logger, eventHandler: EventHandler, navigationClient: INavigationClient, apiId: ApiId, performanceClient: IPerformanceClient, provider: NativeMessageHandler, accountId: string, correlationId?: string) {
+        super(config, browserStorage, browserCrypto, logger, eventHandler, navigationClient, performanceClient, provider, correlationId);
+        this.apiId = apiId;
+        this.accountId = accountId;
+        this.nativeMessageHandler = provider;
+    }
+
+    /**
+     * Acquire token from native platform via browser extension
+     * @param request
+     */
+    async acquireToken(request: PopupRequest|SilentRequest|SsoSilentRequest): Promise<AuthenticationResult> {
+        this.logger.trace("NativeInteractionClient - acquireToken called.");
+        const nativeRequest = await this.initializeNativeRequest(request);
+
+        const messageBody: NativeExtensionRequestBody = {
+            method: NativeExtensionMethod.GetToken,
+            request: nativeRequest
+        };
+
+        const reqTimestamp = TimeUtils.nowSeconds();
+        const response: object = await this.nativeMessageHandler.sendMessage(messageBody);
+        this.validateNativeResponse(response);
```

Comment:
I would suggest this return a validated, type object:

```suggestion
        const validatedResponse: NativeResponse = this.validateNativeResponse(response);
```

Reply:
Done